src/main/java/be/uantwerpen/minelabs/entity/SubatomicParticleEntity.java

```diff
@@ -127,7 +128,8 @@ protected void onEntityHit(EntityHitResult entityHitResult) {
         super.onEntityHit(entityHitResult);
         // notify bohr plate of collision
         if (entityHitResult.getEntity() instanceof BohrBlueprintEntity bohrBlueprintEntity) {
-            bohrBlueprintEntity.onParticleCollision(this);
+            bohrBlueprintEntity.onParticleCollision(this,
+                    this.getOwner() instanceof ServerPlayerEntity owner ? owner : null);
```

Comment:
Move this to where it is needed. I believe getOwner is public so this should be possible.